redis_collections/lists.py

```diff
@@ -98,16 +98,26 @@ def _get_slice(self, index):
         """Helper for getting a slice."""
         assert isinstance(index, slice)
 
+        # For slices without a step we can use the Redis range function
         def slice_trans(pipe):
-            start, stop = self._recalc_slice(index.start, index.stop)
-            values = pipe.lrange(self.key, start, stop)
-            if index.step:
-                # step implemented by pure Python slicing
-                values = values[::index.step]
+            calc_start, calc_stop = self._recalc_slice(index.start, index.stop)
+            if calc_start == index.stop:
+                return []
+
+            values = pipe.lrange(self.key, calc_start, calc_stop)
             values = [self._unpickle(x) for x in values]
+            pipe.multi()
+            return self._create_new(values, pipe=pipe)
 
+        # Otherwise we'll need to pull the whole list and slice in Python
```

Comment:
I tried to think of a better way to do this, but I don't think there's a way to emulate Python's [behavior](https://docs.python.org/3/library/stdtypes.html#common-sequence-operations):

```
The slice of s from i to j with step k is defined as the sequence of items with index x = i + n*k such that 0 <= n < (j-i)/k.
In other words, the indices are i, i+k, i+2*k, i+3*k and so on, stopping when j is reached (but never including j).
If i or j is greater than len(s), use len(s).
If i or j are omitted or None, they become “end” values (which end depends on the sign of k).
Note, k cannot be zero.
If k is None, it is treated like 1.
```